storage/innobase/handler/ha_innodb.cc

```diff
@@ -15968,6 +15968,12 @@ ha_innobase::extra(
 			log_buffer_flush_to_disk();
 		}
 		break;
+	case HA_EXTRA_ABORT_ALTER_COPY:
+		if (!m_prebuilt->table->skip_alter_undo) break;
+		trx = check_trx_exists(ha_thd());
+		m_prebuilt->table->skip_alter_undo = 0;
+		trx->rollback();
+		break;
```

Comment:
It should be more efficient to test and clear the flag closer together, without any function call in between:
```c++
		if (m_prebuilt->table->skip_alter_undo) {
			m_prebuilt->table->skip_alter_undo = 0;
			trx = check_trx_exists(ha_thd());
			trx->rollback();
		}
		break;
```